astropy/time/core.py

```diff
@@ -1336,14 +1340,21 @@ def __sub__(self, other):
                 if other.scale not in (out.scale, None):
                     other = getattr(other, out.scale)
             else:
-                out._set_scale(other.scale if other.scale is not None
-                               else 'tai')
+                if other.scale is not None and self.scale not in TIME_TYPES[other.scale]:
+                    raise TypeError("Cannot subtract Time and TimeDelta instances with scales '{0}' and '{1}'".format(self.scale, other.scale))
+                else:
+                    out._set_scale(other.scale if other.scale is not None else 'tai')
             # remove attributes that are invalidated by changing time
             for attr in ('_delta_ut1_utc', '_delta_tdb_tt'):
                 if hasattr(out, attr):
                     delattr(out, attr)
 
         else:  # T - T
+
+            # the scales should be compatible (e.g., cannot convert TDB to LOCAL)
+            if(self.scale is not None and self.scale not in other.SCALES or
```

Comment:
For `Time`, it should not be possible for `self.scale` or `other.scale` to be `None`. Also, the sets are symmetric, so one only needs to check that `other.scale in self.SCALES`.